client/src/Admin/SitesForm.js

```diff
@@ -10,30 +10,33 @@ function SitesForm() {
   const navigate = useNavigate();
   const user = useAuthContext();
   const { id } = useParams();
-  const [partners, setPartner] = useState(null);
-  const [selectedPartner, setSelectedPartner] = useState(null);
+  const [partners, setPartners] = useState(null);
   const [data, setData] = useState({
     name: '',
     address: '',
-    phonenumber: '',
+    phoneNumber: '',
     email: '',
     website: '',
+    NutritionPartnerIds: [],
   });
 
   useEffect(() => {
     if (id) {
-      Api.sites.get(id).then((response) => setData(response.data));
+      Api.sites.get(id).then((response) => {
+        const { data } = response;
+        data.NutritionPartnerIds = data.NutritionPartners?.map((np) => np.id);
+        setData(data);
```

Comment:
After removing the custom toJSON() function, I move the conversion from full NutritionPartner objects to a NutritionPartnerIds array to the client here...